modules/features/podcasts/src/test/java/au/com/shiftyjelly/pocketcasts/podcasts/viewmodel/podcast/SimilarPodcastHandlerTest.kt

```diff
@@ -0,0 +1,104 @@
+import au.com.shiftyjelly.pocketcasts.repositories.lists.ListRepository
+import au.com.shiftyjelly.pocketcasts.repositories.podcast.PodcastManager
+import au.com.shiftyjelly.pocketcasts.servers.model.DiscoverPodcast
+import au.com.shiftyjelly.pocketcasts.servers.model.ListFeed
+import au.com.shiftyjelly.pocketcasts.servers.server.ListWebService
+import au.com.shiftyjelly.pocketcasts.sharedtest.InMemoryFeatureFlagRule
+import au.com.shiftyjelly.pocketcasts.utils.featureflag.Feature
+import au.com.shiftyjelly.pocketcasts.utils.featureflag.FeatureFlag
+import io.reactivex.Flowable
+import io.reactivex.Single
+import java.util.UUID
+import kotlinx.coroutines.test.runTest
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.whenever
+
+class SimilarPodcastHandlerTest {
+
+    @get:Rule
+    val featureFlagRule = InMemoryFeatureFlagRule()
+
+    private lateinit var listWebService: ListWebService
+    private lateinit var podcastManager: PodcastManager
+    private lateinit var similarPodcasts: SimilarPodcastHandler
+    private lateinit var testPodcast: Podcast
+    private lateinit var testListFeed: ListFeed
+    private lateinit var testDiscoverPodcasts: List<DiscoverPodcast>
+
+    @Before
+    fun setup() {
+        listWebService = mock()
+        val listRepository = ListRepository(listWebService = listWebService, syncManager = null, platform = "android")
+        podcastManager = mock()
+        similarPodcasts = SimilarPodcastHandler(listRepository, podcastManager)
+
+        testPodcast = Podcast(uuid = UUID.randomUUID().toString())
+        testDiscoverPodcasts = listOf(
+            DiscoverPodcast(uuid = UUID.randomUUID().toString(), title = "Test Podcast 1", url = null, author = null, category = null, description = null, language = null, mediaType = null),
+            DiscoverPodcast(uuid = UUID.randomUUID().toString(), title = "Test Podcast 2", url = null, author = null, category = null, description = null, language = null, mediaType = null),
+        )
+        testListFeed = ListFeed(title = "Test List Feed", podcasts = testDiscoverPodcasts, subtitle = null, description = null, date = null, episodes = null, collectionImageUrl = null, featureImage = null, headerImageUrl = null, tintColors = null, collageImages = null, webLinkUrl = null, webLinkTitle = null, promotion = null)
+    }
+
+    @Test
+    fun testFeatureFlagDisabledNoPodcastsFetched() {
+        // feature flag is disabled
+        FeatureFlag.setEnabled(feature = Feature.RECOMMENDATIONS, enabled = false)
+
+        similarPodcasts.setEnabled(true)
+
+        val list = similarPodcasts.getSimilarPodcastListFlowable(testPodcast)
+            .test()
+            .awaitCount(1)
+            .assertNoErrors()
+            .values()
+            .first()
+
+        // no podcasts should be found
+        assert(list is SimilarPodcastsResult.Empty)
```

Comment:
These `assert(x is Foo)` calls should be `org.junit.Assert.assertTrue` instead.

Reply:
Thanks, I have changed this too.